src/encoded/static/components/schema.js

```diff
@@ -471,38 +386,117 @@ DisplayObject.defaultProps = {
 };
 
 
-// Display a complete raw schema object.
-class DisplayRawObject extends React.Component {
-    constructor() {
-        super();
+/**
+ * Display the copy-JSON controls at the top of raw schema displays.
+ */
+const RawObjectControls = ({ rawSchemaElement, textHasSelection }) => {
+    const copyHandler = () => {
+        // Select all raw schema text in preparation for copying it, but only if the user hasn't
+        // manually selected raw schema text.
+        // https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard#Using_execCommand()
+        if (rawSchemaElement && !textHasSelection) {
+            const selection = window.getSelection();
+            const range = document.createRange();
+            range.selectNodeContents(rawSchemaElement);
+            selection.removeAllRanges();
+            selection.addRange(range);
+        }
 
-        // Set object properties.
-        this.editor = null;
-    }
+        // Execute copy command. Firefox can throw errors on rare occasion. As this is so unusual,
+        // we just show a console warning in that case.
+        try {
+            document.execCommand('copy');
+        } catch (err) {
+            console.warn('Text copy failed.');
+        }
 
-    componentDidMount() {
-        // Create a read-only brace editor to display the formatted JSON.
-        createJsonDisplay('raw-schema', this.props.schema).then((editor) => {
-            // The brace editor successfully created. Save the reference to the brace editor
-            // instance so we can get rid of it when the JSON display is closed.
-            this.editor = editor;
-        });
-    }
+        // Deselect text if the user hasn't manually selected raw schema text.
+        if (!textHasSelection) {
+            const selection = window.getSelection();
+            selection.removeAllRanges();
+        }
+    };
+
+    return (
+        <div className="raw-schema__controls">
+            <button className="btn btn-info btn-xs" onClick={copyHandler}>
+                {textHasSelection ? <>Copy selected JSON</> : <>Copy all JSON</>}
+            </button>
+        </div>
+    );
+};
```

Comment:
I thought about highlighting the text when you copied (partly because I accidentally had the text selected when you clicked Copy at first) but thought that might be more confusing (now it’s selected, does that mean I should copy it with command-C, or did it already copy?). But I notice when you use the copy button on Github to clone a repo, it doesn’t do nor say anything, so I decided to go with that. Not that Github is necessarily right, but I figure they might do user testing. I don’t know.

Reply:
I guess you could add a little line of text under the button saying "JSON object copied" or "JSON selection copied" but I'm not sure this is very important.